utility/unit_of_work/uow_replica.go

```diff
@@ -0,0 +1,26 @@
+package unit_of_work
+
+import (
+	"github.com/pokt-network/pocket/logger"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+var (
+	_ modules.UtilityUnitOfWork        = &replicaUtilityUnitOfWork{}
+	_ modules.ReplicaUtilityUnitOfWork = &replicaUtilityUnitOfWork{}
+)
+
+type replicaUtilityUnitOfWork struct {
+	baseUtilityUnitOfWork
+}
+
+func NewForReplica(height int64, readContext modules.PersistenceReadContext, rwPersistenceContext modules.PersistenceRWContext) *replicaUtilityUnitOfWork {
```

Comment:
NIT: `readContext` and `rwPersistenceContext`.

I'd say it's outside the scope of this PR but we should probably rename to `readContext` and `writeContext` and have it be implicit that `write` can also read. I don't see us ever having linux like permissions where you can write but can't read.

Maybe add a TODO?

Reply:
I have added this to https://github.com/pokt-network/pocket/issues/406 scope